algoliasearch-common/src/main/java/com/algolia/search/AsyncObjects.java

```diff
@@ -200,6 +200,80 @@
     return getApiClient().saveObjects(getName(), objects, requestOptions);
   }
 
+  /**
+   * Replace all objects
+   *
+   * @param objects the list objects to update
+   * @param safe run the method in a safe way
+   */
+  default void replaceAllObjects(@Nonnull Iterable<T> objects, boolean safe)
+      throws ExecutionException, InterruptedException {
+    replaceAllObjects(objects, safe, new RequestOptions());
+  }
+
+  /**
+   * Replace all objects
+   *
+   * @param objects the list objects to update
+   * @param safe run the method in a safe way
+   * @param requestOptions Options to pass to this request
+   */
+  default void replaceAllObjects(
+      @Nonnull Iterable<T> objects, boolean safe, @Nonnull RequestOptions requestOptions)
+      throws ExecutionException, InterruptedException {
+
+    String tmpName = this.getName() + "_tmp_" + UUID.randomUUID().toString();
+    List<String> scope = new ArrayList<>(Arrays.asList("rules", "settings", "synonyms"));
+    List<AsyncTaskSingleIndex> batchResponses = new ArrayList<>();
+
+    // Copy all index resources from production index
+    AsyncTask copyIndexResponse =
+        getApiClient().copyIndex(getName(), tmpName, scope, requestOptions).get();
+
+    if (safe) {
+      getApiClient().waitTask(copyIndexResponse);
+    }
+
+    // Send records (batched automatically)
+    ArrayList<T> records = new ArrayList<>();
+
+    for (T object : objects) {
+
+      if (records.size() == 1000) {
+        AsyncTaskSingleIndex batchResponse =
+            getApiClient().saveObjects(tmpName, records, requestOptions).get();
+        if (safe) {
+          batchResponses.add(batchResponse);
+        }
+        records.clear();
+      }
+
+      records.add(object);
+    }
+
+    if (records.size() > 0) {
+      AsyncTaskSingleIndex batchResponse =
+          getApiClient().saveObjects(tmpName, records, requestOptions).get();
+      batchResponses.add(batchResponse);
+    }
+
+    // if safe waits for task to finish
+    if (safe) {
+      for (AsyncTaskSingleIndex response : batchResponses) {
+        getApiClient().waitTask(response);
+      }
+    }
+
+    // Move temporary index to production
```

Comment:
Careful with the term `production` here. You may want to replace it by something else.